File: sdk/python/spear/hostcalls/tools.py
""" This file contains the relevant dataclasses and types for tools related hostcalls.
"""

from dataclasses_json import dataclass_json
from dataclasses import dataclass

@dataclass_json
@dataclass
class NewToolParams:
    """
    The parameters for the newtool hostcall.
    """
    name: str
    type: str
    description: str
    required: bool

@dataclass_json
@dataclass
class NewToolRequest:
    """
    The request object for the newtool hostcall.
    """
    name: str
    description: str
    params: list[NewToolParams]
    cb: str

@dataclass_json
@dataclass
class NewToolResponse:
    """
    The response object for the newtool hostcall.
    """
    tool_id: str

@dataclass_json
@dataclass
class NewToolsetRequest:
    """
    The request object for the newtoolset hostcall.
    """
    name: str
    description: str
    tool_ids: list[str]


@dataclass_json
@dataclass
class NewToolsetResponse:
    """
    The response object for the newtoolset hostcall.
    """
    toolset_id: str

@dataclass_json
@dataclass
class ToolsetInstallBuiltinsRequest:
    """
    The request object for the toolset.install_builtins hostcall.
    """
    toolset_id: str

@dataclass_json
@dataclass
class ToolsetInstallBuiltinsResponse:
    """
    The response object for the toolset.install_builtins hostcall.
    """
    toolset_id: str

File: pkg/rpc/payload/tools.go
package payload

import "encoding/json"

type NewToolParam struct {
	Name        string `json:"name"`
	Type        string `json:"type"`
	Description string `json:"description"`
	Required    bool   `json:"required"`
}

type NewToolRequest struct {
	Name        string         `json:"name"`
	Description string         `json:"description"`
	ToolsetID   int            `json:"toolset_id"`
	Params      []NewToolParam `json:"params"`
	Cb          string         `json:"cb"`
}

type NewToolResponse struct {
	ToolsetID int `json:"tool_id"`
}

type NewToolsetRequest struct {
	Name         string `json:"name"`
	Description  string `json:"description"`
	WorkloadName string `json:"workload_name"`
}

type NewToolsetResponse struct {
	ToolsetID int `json:"toolset_id"`
}

type ToolsetInstallBuiltinsRequest struct {
	ToolsetID int `json:"toolset_id"`
}

type ToolsetInstallBuiltinsResponse struct {
	ToolsetID int `json:"toolset_id"`
}

type ToolCallRequest struct {
	ToolsetID int                    `json:"toolset_id"`
	ToolID    int                    `json:"tool_id"`
	Params    map[string]interface{} `json:"params"`
}

func (r *NewToolRequest) Marshal() ([]byte, error) {
	return json.Marshal(r)
}

func (r *NewToolResponse) Marshal() ([]byte, error) {
	return json.Marshal(r)
}

func (r *NewToolsetRequest) Marshal() ([]byte, error) {
	return json.Marshal(r)
}

func (r *NewToolsetResponse) Marshal() ([]byte, error) {
	return json.Marshal(r)
}

func (r *NewToolRequest) Unmarshal(data []byte) error {
	return json.Unmarshal(data, r)
}

func (r *NewToolResponse) Unmarshal(data []byte) error {
	return json.Unmarshal(data, r)
}

func (r *NewToolsetRequest) Unmarshal(data []byte) error {
	return json.Unmarshal(data, r)
}

func (r *NewToolsetResponse) Unmarshal(data []byte) error {
	return json.Unmarshal(data, r)
}

func (r *ToolsetInstallBuiltinsRequest) Marshal() ([]byte, error) {
	return json.Marshal(r)
}

func (r *ToolsetInstallBuiltinsResponse) Marshal() ([]byte, error) {
	return json.Marshal(r)
}

func (r *ToolsetInstallBuiltinsRequest) Unmarshal(data []byte) error {
	return json.Unmarshal(data, r)
}

func (r *ToolsetInstallBuiltinsResponse) Unmarshal(data []byte) error {
	return json.Unmarshal(data, r)
}
